soh/src/code/z_actor.c

```diff
@@ -2182,7 +2185,19 @@ void func_8002F7A0(PlayState* play, Actor* actor, f32 arg2, s16 arg3, f32 arg4)
 }
 
 void func_8002F7DC(Actor* actor, u16 sfxId) {
-    Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    if (actor->id != ACTOR_PLAYER || sfxId < NA_SE_VO_LI_SWORD_N || sfxId > NA_SE_VO_LI_ELECTRIC_SHOCK_LV_KID) {
+        Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &D_801333E0 , &D_801333E0, &D_801333E8);
+    }
+    else {
+        freqMultiplier = CVarGetFloat("gLinkVoiceFreqMultiplier", 1.0);
+        if (freqMultiplier <= 0) freqMultiplier = 1;
+        if (freqMultiplier != 1) {
+            Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &freqMultiplier, &D_801333E0, &D_801333E8);
+        }
+        else {
+            Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &D_801333E0 , &D_801333E0, &D_801333E8);
+        }
```

Comment:
i actually prefer always using `freqMultiplier` to doing a float compare here (see <https://stackoverflow.com/questions/4682889/is-floating-point-ever-ok> for more details on my float comparison hesitations)

since the user facing behavior should be identical regardless, i think a good path forward here is

```suggestion
        freqMultiplier = CVarGetFloat("gLinkVoiceFreqMultiplier", 1.0);
        if (freqMultiplier <= 0) { 
            freqMultiplier = 1;
        }
        // Authentic behavior uses D_801333E0 for both freqScale and a4
        // Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &D_801333E0 , &D_801333E0, &D_801333E8);
        Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &freqMultiplier, &D_801333E0, &D_801333E8);
```

this leaves the original code in a place where future devs can find it easily, and provides some context to the change